### PR TITLE
cilium: encryption, add encrypt-node option to encrypt node traffic

### PR DESCRIPTION
**Pull request by jrfastab**
Adds an option --encrypt-node used to encrypt traffic from nodeIP, e.g. the IP associated with encrypt-interface.

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/8126)

### CODE REVIEW

**Comment by jrfastab** [comment]
todo:  
 (a) disallow with tunnel mode
 (b) rebase
 (c)  test a couple more paths.

**Comment by jrfastab** [comment]
test-me-please

**Comment by jrfastab** [comment]
Pushed latest with rebase, a couple follow patches are needed but don't think they need to be in this series. These are tunnel mode support, patch to support node to pod encryption (its a small patch), optimization for 4.14+ kernels to avoid second pass through stack.

File: Documentation/cmdref/cilium-agent.md [hunk 1]
```diff
@@ -54,6 +54,7 @@ cilium-agent [flags]
       --enable-policy string                       Enable policy enforcement (default "default")
       --enable-tracing                             Enable tracing while determining policy (debugging)
       --encrypt-interface string                   Transparent encryption interface
+      --encrypt-node                               Encrypt node IP
```
Comment (aanm) [inline Documentation/cmdref/cilium-agent.md]: "Encrypt node IP" seems too abstract. Can you write it differently?

**Review by aanm — CHANGES_REQUESTED** [review]

File: Documentation/cmdref/cilium-agent.md (hunk 1 above)
Comment (jrfastab) [reply Documentation/cmdref/cilium-agent.md]: Sure all though I had a bit of trouble with the one-liner explanation, How about,

"Enables encrypting traffic from non-Cilium pods and host networking"

**Comment by jrfastab** [comment]
test-me-please

**Review by joestringer — APPROVED** [review]

**Comment by jrfastab** [comment]
test-me-please

**Comment by jrfastab** [comment]
failed but all we did here is update daemon.sha, 

https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Validated/12715/

**Comment by jrfastab** [comment]
test-me-please

**Comment by jrfastab** [comment]
test-me-please

**Comment by jrfastab** [comment]
only change from last passing run is a sha update.

File: Documentation/cmdref/cilium-agent.md (hunk 1 above)
Comment (aanm) [reply Documentation/cmdref/cilium-agent.md]: @jrfastab yep that sounds good!

**Comment by jrfastab** [comment]
test-me-please

**Review by aanm — APPROVED** [review]

**Pull request merged**